solidity/contracts/cross-chain/wormhole/L1BTCDepositorNttWithExecutor.sol

```diff
@@ -0,0 +1,648 @@
+        require(
+            executorArgs.signedQuote.length > 0,
+            "Real signed quote from Wormhole Executor API is required"
+        );
+        _validateSignedQuote(executorArgs.signedQuote);
+
+        // Store the parameters for use in finalizeDeposit
+        storedExecutorArgs = executorArgs;
+        storedFeeArgs = feeArgs;
+        executorParametersSet = true;
+
+        emit ExecutorParametersSet(
+            msg.sender,
+            executorArgs.signedQuote.length,
+            executorArgs.value
+        );
+    }
+
+    /// @notice Quotes the cost for a specific destination chain using stored executor parameters
+    /// @param _destinationChain Wormhole chain ID of the destination
+    /// @return cost Total cost for the transfer to the specified chain
+    function quoteFinalizeDeposit(uint16 _destinationChain)
+        external
+        view
+        returns (uint256 cost)
+    {
+        require(
+            executorParametersSet,
+            "Must call setExecutorParameters() first with real signed quote"
+        );
+        require(
+            supportedChains[_destinationChain],
+            "Destination chain not supported"
+        );
+
+        return
+            nttManagerWithExecutor.quoteDeliveryPrice(
+                underlyingNttManager,
+                _destinationChain,
+                "", // Empty transceiver instructions for basic transfer
+                storedExecutorArgs,
+                storedFeeArgs
+            );
+    }
+
+    /// @notice Clears stored executor parameters
+    /// @dev Called automatically after successful transfer or can be called manually
+    function clearExecutorParameters() external {
+        delete storedExecutorArgs;
+        delete storedFeeArgs;
+        executorParametersSet = false;
+    }
+
+    /// @notice Checks if executor parameters have been set
+    /// @return isSet True if executor parameters are set and ready for finalizeDeposit
+    function areExecutorParametersSet() external view returns (bool isSet) {
+        return executorParametersSet;
+    }
+
+    /// @notice Gets the stored executor value (for informational purposes)
+    /// @return value The executor value in wei, or 0 if not set
+    function getStoredExecutorValue() external view returns (uint256 value) {
+        return executorParametersSet ? storedExecutorArgs.value : 0;
+    }
+
+    /// @notice Transfers tBTC using NTT Manager With Executor for automatic destination execution
+    /// @dev Uses stored executor parameters set via setExecutorParameters()
+    /// @param amount Amount of tBTC to transfer
+    /// @param destinationChainReceiver Encoded receiver data with chain ID and recipient
+    function _transferTbtc(uint256 amount, bytes32 destinationChainReceiver)
+        internal
+        override
+    {
+        require(
+            executorParametersSet,
+            "Must call setExecutorParameters() first with real signed quote"
+        );
```

Comment:
This could also be a custom error, we repeat this message in other places

Reply:
solver here :https://github.com/threshold-network/tbtc-v2/pull/904/commits/4d22590824f2e3020dae69ba803705216ebcea7c